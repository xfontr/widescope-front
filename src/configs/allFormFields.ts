import {
  FormComponents,
  FormFields,
} from "../components/RenderForm/RenderFormTypes";

export interface FormValue {
  type?: React.HTMLInputTypeAttribute;
  label: string;
  name: FormFields;
  belongsTo: FormComponents[];
  placeholder?: string;
  customClass?: string;
  customGroupClass?: string;
  renderAs?: "input" | "textarea";
}

export const allFormFields: Record<FormFields, FormValue> = {
  name: {
    type: "text",
    label: "Name",
    name: "name",
    belongsTo: ["logIn", "signUp"],
    placeholder: "John Doe",
    customClass: "form__input--user",
  },

  password: {
    type: "password",
    label: "Password",
    name: "password",
    belongsTo: ["logIn", "signUp"],
    customClass: "form__input--password",
  },

  repeatPassword: {
    type: "password",
    label: "Repeat password",
    name: "repeatPassword",
    belongsTo: ["signUp"],
  },

  email: {
    type: "email",
    label: "Email address",
    name: "email",
    belongsTo: ["signUp"],
  },

  projectName: {
    type: "text",
    label: "Name",
    name: "name",
    placeholder: "Music app",
    belongsTo: ["createProject", "updateProject"],
  },

  repository: {
    type: "text",
    label: "Repository URL",
    name: "repository",
    placeholder: "http://www.github.com/music-app",
    belongsTo: ["createProject", "updateProject"],
  },

  logo: {
    type: "file",
    label: "Project logo",
    name: "logo",
    belongsTo: ["createProject"],
    customGroupClass: "area",
  },

  logoUpdate: {
    type: "file",
    label: "Project logo",
    name: "logoUpdate",
    belongsTo: ["updateProject"],
    customGroupClass: "area",
  },

  technologyFront: {
    type: "text",
    label: "Frontend main library or framework",
    name: "technologyFront",
    placeholder: "React",
    belongsTo: ["createProject", "updateProject"],
  },

  technologyBack: {
    type: "text",
    label: "Backend main library or framework",
    name: "technologyBack",
    placeholder: "React",
    belongsTo: ["createProject", "updateProject"],
  },

  description: {
    label: "Description",
    name: "description",
    placeholder: "Music app is about sharing music between users",
    belongsTo: ["createProject", "updateProject"],
    renderAs: "textarea",
    customGroupClass: "area",
  },
};
