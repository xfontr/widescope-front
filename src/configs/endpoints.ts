const endpoints = {
  usersRoot: "/users",
  signUp: "/users/sign-up",
  logIn: "/users/log-in",

  projectsRoot: "/projects",
  getAll: "/projects/all",
  projectById: "/projects/",
  projectsByAuthor: "/projects/author/",
  createProject: "/projects/new",
};

export default endpoints;
