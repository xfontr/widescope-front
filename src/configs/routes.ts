const routes = {
  root: "/",
  home: "/home",
  signUp: "/sign-up",
  logIn: "/log-in",
  explore: "/explore",
  projectDetails: "/project/:projectId",
};

export default routes;
