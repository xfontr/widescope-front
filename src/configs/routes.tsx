import React from "react";
import { Navigate } from "react-router-dom";
import LandingPage from "../pages/LandingPage/LandingPage";
const ExplorePage = React.lazy(
  () => import("../pages/ExplorePage/ExplorePage")
);
const LogInPage = React.lazy(() => import("../pages/LogInPage/LogInPage"));
const ManageProjectPage = React.lazy(
  () => import("../pages/ManageProjectPage/ManageProjectPage")
);
const NotFoundPage = React.lazy(
  () => import("../pages/NotFoundPage/NotFoundPage")
);
const ProjectDetailsPage = React.lazy(
  () => import("../pages/ProjectDetailsPage/ProjectDetailsPage")
);
const SignUpPage = React.lazy(() => import("../pages/SignUpPage/SignUpPage"));
const UserProjectsPage = React.lazy(
  () => import("../pages/UserProjectsPage/UserProjectsPage")
);

export interface NavRoute {
  path: string;
  display: "never" | "always" | "notLogged" | "logged";
  name: string;
  skip?: boolean;
  action?: (action: () => void) => void;
  render?: () => JSX.Element;
}

export const navRoutes: Record<string, NavRoute> = {
  notFound: {
    path: "*",
    display: "never",
    name: "Not found",
    render: () => <NotFoundPage />,
  },

  root: {
    path: "/",
    display: "never",
    name: "Home",
    render: () => <Navigate to={navRoutes.home.path} />,
  },

  home: {
    path: "/home",
    display: "always",
    name: "Home",
    render: () => <LandingPage />,
  },

  explore: {
    path: "/explore",
    display: "always",
    name: "Explore",
    render: () => <ExplorePage />,
  },

  signUp: {
    path: "/sign-up",
    display: "notLogged",
    name: "Sign up",
    render: () => <SignUpPage />,
  },

  logIn: {
    path: "/log-in",
    display: "notLogged",
    name: "Log in",
    render: () => <LogInPage />,
  },

  logOut: {
    path: "",
    display: "logged",
    name: "Log out",
    action: (action) => action(),
  },

  personalProjects: {
    path: "/personal-projects",
    display: "logged",
    name: "Your projects",
    render: () => <UserProjectsPage />,
  },

  createProject: {
    path: "/project/new",
    display: "logged",
    skip: true,
    name: "Post a project",
    render: () => <ManageProjectPage isCreate={true} />,
  },

  updateProject: {
    path: "/projects/update/:projectId",
    display: "never",
    name: "Update project",
    render: () => <ManageProjectPage isCreate={false} />,
  },

  projectDetails: {
    path: "/project/:projectId",
    display: "never",
    name: "Project details",
    render: () => <ProjectDetailsPage />,
  },
};
