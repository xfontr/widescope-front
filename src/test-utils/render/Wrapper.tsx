import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { store } from "../../app/store";
import GlobalStyles from "../../styles/GlobalStyles";
import styledMainTheme from "../../styles/styledMainTheme";

interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return (
    <ThemeProvider theme={styledMainTheme}>
      <Provider store={store}>
        <BrowserRouter>
          <GlobalStyles />
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>
  );
};

export default Wrapper;
