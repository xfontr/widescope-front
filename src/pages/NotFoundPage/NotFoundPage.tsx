import { useNavigate } from "react-router-dom";
import Button from "../../components/Button/Button";
import { navRoutes } from "../../configs/routes";

const NotFoundPage = (): JSX.Element => {
  const navigate = useNavigate();

  return (
    <>
      <h2 className="page__title page__title--wide">
        Oops! We couldn't find what you are
        <span className="page__title--bold"> looking for</span>
      </h2>
      <span className="page__title-subheading">
        But hey, what about exploring some of our projects?
      </span>
      <Button
        children="Explore"
        type="button"
        action={() => {
          navigate(navRoutes.explore.path);
        }}
      />
    </>
  );
};

export default NotFoundPage;
