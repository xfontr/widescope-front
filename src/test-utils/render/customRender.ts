import { render } from "@testing-library/react";
import { Wrapper } from "./Wrapper";

const customRender = (view: JSX.Element, ...options: any) =>
  render(view, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
