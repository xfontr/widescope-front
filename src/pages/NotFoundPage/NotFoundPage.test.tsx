import { render, screen } from "../../test-utils/render/customRender";
import userEvent from "@testing-library/user-event";
import NotFoundPage from "./NotFoundPage";
import { navRoutes } from "../../configs/routes";

const mockNavigate = jest.fn().mockReturnThis();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("Given a NotFoundPage component", () => {
  describe("When instantiated", () => {
    test("Then it should display a not found message and a button", () => {
      render(<NotFoundPage />);

      const notFoundPage = [
        screen.getByRole("heading", {
          name: "Oops! We couldn't find what you are looking for",
          level: 2,
        }),
        screen.getByText("But hey, what about exploring some of our projects?"),
        screen.getByRole("button", { name: "Explore" }),
      ];

      notFoundPage.forEach((element) => {
        expect(element).toBeInTheDocument();
      });
    });

    test("If the user clicks the 'Explore' button, it should call the navigate function", async () => {
      render(<NotFoundPage />);

      const button = screen.getByRole("button", { name: "Explore" });

      await userEvent.click(button);

      expect(mockNavigate).toHaveBeenCalledWith(navRoutes.explore.path);
    });
  });
});
