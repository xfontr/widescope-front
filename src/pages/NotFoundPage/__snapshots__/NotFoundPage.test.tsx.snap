// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a NotFoundPage component When instantiated Then it should match the snapshot 1`] = `
Array [
  <h2
    className="page__title page__title--wide"
  >
    Oops! We couldn't find what you are
    <span
      className="page__title--bold"
    >
       looking for
    </span>
  </h2>,
  <span
    className="page__title-subheading"
  >
    But hey, what about exploring some of our projects?
  </span>,
  <button
    className="sc-bczRLJ jZmJeh undefined button--default "
    onClick={[Function]}
    rel="noopener noreferrer"
    target="_blank"
    type="button"
  >
    Explore
  </button>,
]
`;
